BlueprintUI/Sources/Layout/DecorateBackground.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  DecorateBackground.swift
+//  BlueprintUI
+//
+//  Created by Kyle Van Essen on 11/4/20.
+//
+
+import UIKit
+
+
+///
+/// Places a background behind the given `wrapped` element,
+/// and overflows it by the amount specified in `overflow`.
+///
+/// The size of the element is determined only by the `wrapped` element –
+/// that is, the `background` element will overflow the bounds of the element, but not
+/// affect its layout or size. This is similar to how UIKit handles shadows: They live outside of the layout
+/// rect of the view:
+///
+/// The arrows represent the measured size of the view for layout purposes.
+/// ```
+/// ┌───────────────────┐
+/// │    Background     │
+/// │ ┏━━━━━━━━━━━━━━━┓ │ ▲
+/// │ ┃               ┃ │ │
+/// │ ┃    Wrapped    ┃ │ │
+/// │ ┃               ┃ │ │
+/// │ ┗━━━━━━━━━━━━━━━┛ │ ▼
+/// └───────────────────┘
+///   ◀───────────────▶
+/// ```
+///
+/// This is useful to render on-touch or selected states of elements, where you
+/// want to provide a padded background that does otherwise affect the layout of the element,
+/// which is likely controlled by its parent element.
+///
+public struct DecorateBackground : Element {
+    
+    /// The element which provides the sizing and measurement.
+    public var wrapped : Element
+    
+    /// The element which is used to render the background.
+    /// It is stretched to fit the `wrapped` content, plus the `overflow` padding.
+    ///
+    /// If you have a 100w x 50h element, and an overflow of (10, 10, 10, 10),
+    /// the measured sized will be 100w x 50h, and the background will be
+    /// sized to be 120w x 70h.
+    /// ```
+    /// ┌───────────────────┐
+    /// │ ┏━━━━━━━━━━━━━━━┓ │ ▲
+    /// │ ┃               ┃ │ │
+    /// │ ┃    Wrapped    ┃ │ │
+    /// │ ┃               ┃ │ │
+    /// │ ┗━━━━━━━━━━━━━━━┛ │ ▼
+    /// └───────────────────┘
+    ///   ◀───────────────▶
+    /// ```
+    public var background : Element
+    
+    /// How much the background should overflow the measured bounds of the
+    /// element. Positive values overflow outside of the bounds, and negative
+    /// values underflow to inside the bounds.
+    public var overflow : UIEdgeInsets
```

Comment:
I feel like we might want to make this an enum? insets might not work for every layout you'd want to do with a background - i.e., what if I want a fixed size decoration, positioned on the top right corner behind the wrapped element? Not saying we need to add those all now, but making this an enum now means easier migration

Reply:
Actually; this gives me a better idea. Make this be a `Decoration` element, and you can put the decoration in front of or behind the content, and then also as you describe, more carefully control the layout. This is basically what is described in https://github.com/square/Blueprint/issues/49.

Reply:
I think that makes a lot of sense

Reply:
Done